flask_backend_design.py

```diff
@@ -0,0 +1,272 @@
+                            amount=amount,
+                            transactiontype=transactiontype))
+
+
+""" Update the balance based on Transaction type \
+    and displays the details for the same"""
+
+
+@app.route('/balance/<accountnumber>/<amount>/<transactiontype>')
+def account_balance(accountnumber, amount, transactiontype):
+    if transactiontype == 'deposit':
+        cursor.execute("INSERT INTO Transactions_bankdemo \
+        (accountnumber,Transactiondate,AMount,Transactiontype,Createdat,\
+        Createdby,Modifiedat,Modifiedby) \
+VALUES (?,?,?,?,?,?,?,?)", accountnumber, date, amount,
+                       transactiontype, date, CurrentUser, date, CurrentUser)
+        cursor.commit()
+        cursor.execute("Update AccountBalance_bankdemo SET Balance=balance+?,\
+                        Modifiedat=?,Modifiedby=? Where accountnumber=?",
+                       amount, date, CurrentUser, accountnumber)
+        cursor.commit()
+        cursor.execute("SELECT D.accountnumber,convert(float,d.Amount)Amount,\
+        CONVERT(DATETIME,d.DepositDate)DepositDate,\
+        convert(float,B.Balance)Balance \
+        FROM Deposit_bankdemo d WITH(Nolock)\
+        INNER JOIN AccountBalance_bankdemo B WITH(Nolock)\
+            ON d.accountnumber=B.accountnumber\
+        Where d.accountnumber=? ", accountnumber)
+        rows = cursor.fetchall()
+        if rows:
+            for row in rows:
+                return jsonify({"accountnumber": row[0],
+                                "DepositedAmount": row[1],
+                                "DepositDate": row[2],
+                                "Balance": row[3]})
+
+    if transactiontype == 'withdrawl':
+        cursor.execute("INSERT INTO Transactions_bankdemo \
+        (accountnumber,Transactiondate,AMount,Transactiontype,Createdat,\
+        Createdby,Modifiedat,Modifiedby) \
+        VALUES (?,?,?,?,?,?,?,?)", accountnumber, date, amount,
+                       transactiontype, date, CurrentUser, date, CurrentUser)
+        cursor.commit()
+        cursor.execute("Update AccountBalance_bankdemo SET Balance=balance-?,\
+        Modifiedat=?,Modifiedby=?\
+        Where accountnumber=?", amount, date, CurrentUser, accountnumber)
+        cursor.commit()
+        cursor.execute("SELECT D.accountnumber,convert(float,d.Amount)Amount,\
+        CONVERT(DATETIME,d.WithdrawlDate)WithdrawlDate,\
+        convert(float,B.Balance)Balance \
+        FROM Withdrawl_bankdemo d WITH(Nolock)\
+        INNER JOIN AccountBalance_bankdemo B WITH(Nolock)\
+            ON d.accountnumber=B.accountnumber\
+        Where d.accountnumber=? ", accountnumber)
+        rows = cursor.fetchall()
+        if rows:
+            for row in rows:
+                return jsonify(
+                    {"accountnumber": row[0],
+                     "WithdrawlAmount": row[1],
+                     "WithdrawlDate": row[2],
+                     "Balance": row[3]})
+
```

Comment:
Very big method. A method should not contain more than 15 - 20 lines (except few exceptional cases.)
This can be broken down to individual method.